diff --git a/Makefile.include b/Makefile.include
index f24507a..fcf80fd 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -116,6 +116,22 @@ else
   include $(target_makefile)
 endif
 
+### Verbosity control. Use  make V=1  to get verbose builds.
+
+CC_normal	:= $(CC) #-mfpu=fpv4-sp-d16 -mfloat-abi=softfp
+AR_normal	:= $(AR)
+
+CC_quiet	= @echo "  CC       " $< && $(CC_normal)
+AR_quiet	= @echo "  AR       " $@ && $(AR_normal)
+
+ifeq ($(V),1)
+  CC		= $(CC_normal)
+  AR		= $(AR_normal)
+else
+  CC		= $(CC_quiet)
+  AR		= $(AR_quiet)
+endif
+
 ### Forward comma-separated list of arbitrary defines to the compiler
 
 COMMA := ,
